### PR TITLE
fix: refactor implementation and keep stable interface

### PR DESCRIPTION
**Pull request by spirosikmd**
This is a refactor proposal that will make it easier to understand and
develop further this library. It uses a configuration for the endpoints
and the `axios` library for the HTTP requests instead of the custom
implementation that relied on the `http` and `https` node modules.

This version keeps the public interface the same to not break backwards
compatibility and has feature parity with the existing implementation.

It also fixes an issue with processing requests that have more items and
iteration should be used.

### CODE REVIEW

File: lib/action.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const request = require('./request');
+const { sign } = require('./signing');
+const ClientError = require('./error');
+
+// This doesn't really make sense anymore as the version won't change (0.0.0-development)
```
Comment (tzengerink) [inline lib/action.js]: Is this something we want to change now? Is there a ticket for this?

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+function createClient(accessKey, privateKey, options) {
+  const apiOptions = parseApiOptions(options);
+
+  const actions = {};
+
+  createActions(actions, endpoints, apiOptions, accessKey, privateKey);
+
+  return actions;
```
Comment (tzengerink) [inline lib/index.js]: How about having `createActions` return the created `actions`?

```js
function createClient(accessKey, privateKey, options) {
    const apiOptions = parseApiOptions(options);
    return createActions({}, endpoints, apiOptions, accessKey, privateKey);
}
```

File: lib/action.js (hunk 1 above)
Comment (spirosikmd) [reply lib/action.js]: In my opinion, if we don't want to deal with this now we should remove it. This data now doesn't make sense. I assume that we want to keep this functionality. But if we don't want to deal with this now, we should create an issue to track it. @shuisman what do you think?

Comment (spirosikmd) [reply lib/action.js]: This is not needed anymore, so we can remove it!

**Pull request merged**